Zend/zend_API.h

```diff
@@ -1879,6 +1902,26 @@ static zend_always_inline int zend_parse_arg_str_or_long(zval *arg, zend_string
 	return 1;
 }
 
+static zend_always_inline int zend_parse_arg_str_or_object(
+	zval *arg, zend_string **dest_str, zval **dest_object, int allow_null
```

Comment:
We might be better of masking this STR_OR_OBJ and returning zend_object* here.

Reply:
Using `zend_object*` is definitely fine for me, but I'd like to stick with the `STR_OR_OBJECT` name because all other object-related fast ZPP macros use the full name (e.g. Z_PARAM_OBJECT_*) + TBH I don't really like that there are so many abbreviations in the code, they made even more difficult for me to understand what's going on.

Reply:
The OBJ suggestion is specifically to distinguish it from the existing OBJECT, which uses zval* instead of zend_object*. Similar to how we distinguish STR (zend_string*) and STRING (const char*).

Reply:
I'm also open to some other naming convention, I just think it should be different from OBJECT.

Reply:
OK, I see your point, I'll change it then. It was a bit surprising for me that `Z_PARAM_OBJECT` returns a `zval*` (and It's the reason I was originally doing the same).